Generator/Sources/NeedleFramework/Parsing/Tasks/BaseFileFilterTask.swift

```diff
@@ -38,37 +38,50 @@ class FileFilterTask: AbstractTask<FilterResult> {
     /// - parameter exclusionPaths: The list of path components to check.
     /// If the given URL's path contains any elements in this list, the
     /// URL will be excluded.
-    init(url: URL, exclusionSuffixes: [String], exclusionPaths: [String]) {
+    /// - parameter taskId: The tracking task ID to use.
+    init(url: URL, exclusionSuffixes: [String], exclusionPaths: [String], taskId: TaskIds) {
         self.url = url
         self.exclusionSuffixes = exclusionSuffixes
         self.exclusionPaths = exclusionPaths
-        super.init(id: TaskIds.fileFilterTask.rawValue)
+        super.init(id: taskId.rawValue)
     }
 
     /// Execute the task and returns the filter result indicating if the file
     /// should be parsed.
     ///
     /// - returns: The `FilterResult`.
     /// - throws: Any error occurred during execution.
-    override func execute() throws -> FilterResult {
+    final override func execute() throws -> FilterResult {
         let urlFilter = UrlFilter(url: url, exclusionSuffixes: exclusionSuffixes, exclusionPaths: exclusionPaths)
         if !urlFilter.filter() {
             return FilterResult.skip
         }
 
         let content = try? String(contentsOf: url)
         if let content = content {
-            let contentFilter = ContentFilter(content: content)
-            if contentFilter.filter() {
-                return FilterResult.shouldParse(url, content)
-            } else {
-                return FilterResult.skip
+            let filters = self.filters(for: content)
+            for filter in filters {
```

Comment:
nit: another good opportunity for a Sequence function: `first(where:)` would work well here.

Reply:
Will do.